public/app/trustedTypePolicies.ts

```diff
@@ -0,0 +1,13 @@
+import DOMPurify from 'dompurify';
+
+import { textUtil } from '@grafana/data';
+import { config } from '@grafana/runtime';
+
+if (config.bootData.settings.featureToggles.trustedTypes && window.trustedTypes && window.trustedTypes.createPolicy) {
+  // check if browser supports Trusted Types
+  window.trustedTypes.createPolicy('default', {
+    createHTML: (string) => DOMPurify.sanitize(string, { RETURN_TRUSTED_TYPE: true }).toString(),
```

Comment:
This returns a string, rather than TrsutedHTML, which createHTML weirdly requires.  Is `RETURN_TRUSTED_TYPE` even needed then?

Reply:
It turns out that the createPolicy handles the type creation. I can see that this is useful in cases where you need to create TrustedHTML without a policy.

Reply:
`createHTML: (string) => DOMPurify.sanitize(string)` is fine since `createHTML` takes a string and returns a TrustedHTML object, and DOMPurify does the same with `RETURN_TRUSTED_TYPE` (if UA supports TT). So basically,  are we doing the same thing twice here. 

> I can see that this is useful in cases where you need to create TrustedHTML without a policy.

Do you perhaps mean the other way around; `createPolicy` will create a dedicated policy. If this dedicated policy is not present, then "default" will be used (if defined). So `DOMPurify.sanitize(string, { RETURN_TRUSTED_TYPE: true })` is useful when you want to return a TrustedHTML object without creating a policy. 

In commit https://github.com/grafana/grafana/pull/64975/commits/45db44ef8d8eea5fc73c77c1f006410c87883a83 you can see `textUtil.sanitizeUrl(string)` will return a string, but the createPolicy will convert it to a TrustedScriptURL object.